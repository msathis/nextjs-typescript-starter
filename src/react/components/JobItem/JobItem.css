.Job {
  margin: 5px 10px 0 5px;
  padding: 5px 10px;
  cursor: pointer;
  display: flex;
  align-items: center;
  justify-content: space-between;
  color: #313039;
  background: #fff;
}
.JobFlags {
  display: flex;
  align-items: center;
  margin: 5px 0;
}
.Company {
  font-weight: bold;
  padding: 10px 0 5px 0;
}
.Location {
  padding: 0 5px;
}
.Country {
  display: flex;
  color: #9f9baf;
}
.Flag {
  border-radius: 5px;
  background-color: orange;
  display: inline-block;
  padding: 0 5px;
  margin: 0 5px;
  color: #fff;
  font-size: 12px;
}
