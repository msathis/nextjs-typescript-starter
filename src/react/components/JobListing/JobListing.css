.Job-Listing {
  flex: 1;
  height: 100%;
  overflow: scroll;
}
