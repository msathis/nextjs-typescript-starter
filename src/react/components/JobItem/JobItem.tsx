import React from "react";
import PropTypes from "prop-types";

import "./JobItem.css";

const propTypes = {
  job: PropTypes.shape({
    title: PropTypes.string.isRequired,
    company: PropTypes.shape({
      name: PropTypes.string.isRequired
    }),
    cities: PropTypes.array,
    remotes: PropTypes.array
  }),
  onClick: PropTypes.func,
  style: PropTypes.object
};
const defaultProps = {};

export default function JobItem({ job, style, onClick }) {
  return (
    <div className="Job" style={style} onClick={onClick}>
      <div>
        <div className="Title">{job.title}</div>
        <div className="Company">{job.company.name}</div>
      </div>
      <div className="JobFlags">
        <div className="Country">
          {job.cities.map(city => (
            <div key={city.name} className="Location">
              <span>{city.name}, </span>
              <span>{city.country.name}</span>
            </div>
          ))}
        </div>
        {job.remotes ? <div className="Flag">Remote</div> : null}
      </div>
    </div>
  );
}

JobItem.propTypes = propTypes;
JobItem.defaultProps = defaultProps;
