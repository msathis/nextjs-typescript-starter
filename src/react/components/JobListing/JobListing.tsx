import React from "react";
import PropTypes from "prop-types";

import "./JobListing.css";
import JobItem from "../JobItem/JobItem";

const propTypes = {
  jobs: PropTypes.arrayOf(
    PropTypes.shape({
      title: PropTypes.string.isRequired,
      company: PropTypes.shape({
        name: PropTypes.string.isRequired
      }),
      cities: PropTypes.array,
      remotes: PropTypes.array
    })
  ),
  onSelect: PropTypes.func.isRequired
};
const defaultProps = {};

export default function JobListing({ jobs, onSelect }) {
  return (
    <div className="Job-Listing">
      {jobs.map(job => (
        <JobItem
          style={{ cursor: "pointer" }}
          onClick={() => onSelect(job)}
          job={job}
          key={job.id}
        />
      ))}
    </div>
  );
}

JobListing.propTypes = propTypes;
JobListing.defaultProps = defaultProps;
