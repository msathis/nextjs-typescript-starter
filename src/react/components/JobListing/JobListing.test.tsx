import React from "react";
import { shallow, configure } from "enzyme";
import Adapter from "enzyme-adapter-react-16";
import { expect as expectChai } from "chai";

import JobListing from "./JobListing";
import JobItem from "../JobItem/JobItem";

configure({ adapter: new Adapter() });

describe("JobListing", () => {
  it("should render component", () => {
    const component = shallow(
      <JobListing
        onSelect={() => {}}
        jobs={[
          {
            id: 1,
            title: "Test Job 1"
          },
          {
            id: 2,
            title: "Test Job 2"
          }
        ]}
      />
    );

    expect(component.getElements()).toMatchSnapshot();
    expectChai(component.find(JobItem)).to.have.lengthOf(2);
  });
});
