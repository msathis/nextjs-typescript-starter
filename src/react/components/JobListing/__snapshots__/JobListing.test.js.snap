// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`JobListing should render component 1`] = `
Array [
  <div
    className="Job-Listing"
  >
    <JobItem
      job={
        Object {
          "id": 1,
          "title": "Test Job 1",
        }
      }
      onClick={[Function]}
      style={
        Object {
          "cursor": "pointer",
        }
      }
    />
    <JobItem
      job={
        Object {
          "id": 2,
          "title": "Test Job 2",
        }
      }
      onClick={[Function]}
      style={
        Object {
          "cursor": "pointer",
        }
      }
    />
  </div>,
]
`;
